source/encoder.c

```diff
@@ -1043,22 +1037,38 @@ CBOREncoder_encode_datetime(CBOREncoderObject *self, PyObject *value)
 static PyObject *
 CBOREncoder_encode_date(CBOREncoderObject *self, PyObject *value)
 {
-    PyObject *datetime, *ret = NULL;
-
-    if (PyDate_Check(value)) {
-        datetime = PyDateTimeAPI->DateTime_FromDateAndTime(
-                PyDateTime_GET_YEAR(value),
-                PyDateTime_GET_MONTH(value),
-                PyDateTime_GET_DAY(value),
-                0, 0, 0, 0, self->tz,
-                PyDateTimeAPI->DateTimeType);
-        if (datetime) {
-            ret = CBOREncoder_encode_datetime(self, datetime);
-            Py_DECREF(datetime);
-            return ret;
+    // semantic type 100 or 1004
+
+    PyObject *tmp, *ret = NULL;
+    const char *buf;
+    Py_ssize_t length;
+    if (value) {
```

Comment:
One more thing: why are you checking `value` here anyway? Shouldn't you trust that the value passed to this function? No other encoder function does this.

Reply:
You are right, value is always non-null here.